recovery.go

```diff
@@ -114,6 +116,17 @@ LOOP:
 						_ = pprof.Lookup("heap").WriteTo(logger, 2)
 						_ = pprof.Lookup("goroutine").WriteTo(logger, 2)
 
+						crashDir := "./crashes"
+						if err := os.MkdirAll(crashDir, 0700); err == nil {
+							crashTimestamp := time.Now().Format("2006-01-02-15-04")
+							crashFile := fmt.Sprintf("%s/heap-%s.pprof", crashDir, crashTimestamp)
+
+							if heapFile, err := os.Create(crashFile); err == nil {
+								defer heapFile.Close()
```

Comment:
Rather than defer, we should manually close right after writing to the heap file.